### ISSUE TITLE
Feature Request:  in-settings filter/sort for the 'Pick a search engine' activity

### ISSUE THREAD

**Issue by Gittalax** [markusfisch/BinaryEye#327]
Since version 1.54.0, the list has outgrown space, I have to scroll to reach my custom URL, which is at the end of the list. In addition, this (for me) is obscured for long seconds after each scan by that notification: 'No application can open that'.
Logcat:
...I ActivityTaskManager: START u0 {act=android.intent.action.VIEW dat=5901576049736} from uid 10193  (uid of BinaryEye)

Personally, I never use Google, Amazon, Ebay, Yandex, while others like these. So I'd like to suggest a submenu in Settings that lists these predefined search engines and maybe the custom URL(s) here, next to a checkbox (maybe a drag sorting too :)).

**Comment by markusfisch** [comment]
Yes, I can totally see that the list got too long and that there should be an editor for that 👍

**Comment by Jsewill** [comment]
Google Product search might be smart to implement in that list. I am not very familiar with Google's various search endpoints, but I suspect something like this might work: https://www.google.com/search?tbm=shop&q=T0F29A

I can open another issue for this if you agree and prefer.

**Comment by markusfisch** [comment]
Good idea! 👍 I will add this with the next release.
